core/src/main/scala/dev/chronicles/core/VersionTracker.scala

```diff
@@ -60,14 +68,19 @@ trait VersionTracker[F[_]] {
     */
   def setCurrentVersion(table: TableName, id: VersionTracker.CommitId): F[Unit]
 
+  /**
+    * Get the flag that indicates whether a table is a snapshot table or not.
+    */
+  def isSnapshotTable(table: TableName): F[Boolean]
```

Comment:
Indeed, a stateful fold that results in either a `TableVersion.Snapshot` or `TableVersion.Partitioned` would probably be simpler. Might have a crack at this to see if it works out nicely.

As for the latter part, that highlights a bit of tension in the overall design I think, as to how to treat snapshot tables vs partitioned ones. That's a bit of an ongoing concern I think.

Maybe one way to simplify it for this case would be to let partitioned tables have a table version as well. That is how it's represented for the real tables; you can specify the root path of a partitioned table even if each of its partitions have a specific path too, it's just not that useful. But maybe it would be cleaner to let that to be expressed in this API too.

But maybe it would be worth taking a step back and considering the whole model for partitioned vs snapshot tables before attempting that.